crates/pathfinder/src/state/contract_hash.rs

```diff
@@ -0,0 +1,419 @@
+use anyhow::{Context, Error, Result};
+use pedersen::StarkHash;
+use serde::Serialize;
+use sha3::Digest;
+
+pub fn compute_contract_hash(contract_definition_dump: &str) -> Result<StarkHash> {
+    use json::EntryPointType::*;
+    // related python impl: https://github.com/starkware-libs/cairo-lang/blob/7712b21fc3b1cb02321a58d0c0579f5370147a8b/src/starkware/starknet/core/os/contract_hash.py
+
```

Comment:
I think a broad overview of the hashing process and oddities would be useful here. Just to get a sense for what is coming / expected.

Reply:
Updated, have a look?